idb/dummy.go

```diff
@@ -217,29 +235,34 @@ type IndexerDb interface {
 
 	LoadGenesis(genesis types.Genesis) (err error)
 	SetProto(version string, proto types.ConsensusParams) (err error)
-	GetProto(version string) (proto types.ConsensusParams, err error)
+	GetProtoTx(tx *sql.Tx, version string) (proto types.ConsensusParams, err error)
 
 	GetImportState() (is *ImportState, err error)
 	SetImportState(ImportState) (err error)
-	GetMaxRoundAccounted() (round uint64, err error)
+	GetMaxRoundAccountedTx(tx *sql.Tx) (round uint64, err error)
 	GetMaxRoundLoaded() (round uint64, err error)
-	GetSpecialAccounts() (SpecialAccounts, error)
+	GetSpecialAccountsTx(ctx context.Context, tx *sql.Tx) (SpecialAccounts, error)
 	GetDefaultFrozen() (defaultFrozen map[uint64]bool, err error)
 
 	// YieldTxns returns a channel that produces the whole transaction stream after some round forward
 	YieldTxns(ctx context.Context, prevRound int64) <-chan TxnRow
 
 	CommitRoundAccounting(updates RoundUpdates, round uint64, blockPtr *types.Block) (err error)
 
-	GetBlock(ctx context.Context, round uint64, options GetBlockOptions) (block types.Block, transactions []TxnRow, err error)
+	GetBlock(ctx context.Context, round uint64,
+		options GetBlockOptions) (block types.Block, transactions []TxnRow, err error)
+	GetBlockTx(ctx context.Context, tx *sql.Tx, round uint64,
+		options GetBlockOptions) (block types.Block, transactions []TxnRow, err error)
+
+	TransactionsTx(ctx context.Context, tx *sql.Tx, tf TransactionFilter) <-chan TxnRow
+	GetAccountsTx(ctx context.Context, tx *sql.Tx, opts AccountQueryOptions) <-chan AccountRow
+	AssetsTx(ctx context.Context, tx *sql.Tx, filter AssetsQuery) <-chan AssetRow
+	AssetBalancesTx(ctx context.Context, tx *sql.Tx, abq AssetBalanceQuery) <-chan AssetBalanceRow
+	ApplicationsTx(tx *sql.Tx, filter *models.SearchForApplicationsParams) <-chan ApplicationRow
 
-	Transactions(ctx context.Context, tf TransactionFilter) <-chan TxnRow
-	GetAccounts(ctx context.Context, opts AccountQueryOptions) <-chan AccountRow
-	Assets(ctx context.Context, filter AssetsQuery) <-chan AssetRow
-	AssetBalances(ctx context.Context, abq AssetBalanceQuery) <-chan AssetBalanceRow
-	Applications(ctx context.Context, filter *models.SearchForApplicationsParams) <-chan ApplicationRow
+	Health(ctx context.Context) (status Health, err error)
 
-	Health() (status Health, err error)
+	MakeTx(ctx context.Context) (*sql.Tx, error)
```

Comment:
Ah, you're right we need some sort of interface around `Begin` in order for the DB interface to work... I hadn't considered that.

We would also need a wrapper around `sql.Tx` to finish the abstraction or else we can only support sql databases.

Given this problem what do you think about changing the interface like this?
```diff
-	Transactions(ctx context.Context, tf TransactionFilter) <-chan TxnRow
+	Transactions(ctx context.Context, tf TransactionFilter) (<-chan TxnRow, currentRound)
```

The API layer can return an error (or retry) if it needs multiple calls and the round doesn't match.